### PR TITLE
MINOR: Improve on reset integration test

### PR DESCRIPTION
**Pull request by guozhangwang**
1. Refactor AbstractResetIntegrationTest with producer / consumer / streams config, along with common client config embedded with ssl configs if necessary.
2. Use specific test id to replace the testNo suffix.
3. In testReprocessingFromScratchAfterResetWithIntermediateUserTopic, additional check on intermediate result.
4. Add the same result for SSL test suite as well.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java [hunk 1]
```diff
@@ -40,271 +42,319 @@
...
-        return null;
+    void cleanupTest() throws Exception {
+        IntegrationTestUtils.purgeLocalStreamsState(STREAMS_CONFIG);
     }
 
-    void testReprocessingFromScratchAfterResetWithoutIntermediateUserTopic() throws Exception {
-        final Properties sslConfig = getClientSslConfig();
-        final Properties streamsConfiguration = prepareTest();
+    private void add10InputElements() throws java.util.concurrent.ExecutionException, InterruptedException {
+        mockTime.sleep(10);
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java]: could we turn this into a loop? i.e., sth-like?
```
List<KeyValue> records = Arrays.asList(...);
for(KeyValue record : records) {
   mockTime.sleep(10);
   IntegrationTestUtils....
}
```

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java [hunk 2]
```diff
@@ -40,271 +42,319 @@
...
+        Assert.assertEquals(1, exitCode);
+    }
+
+    void testReprocessingFromScratchAfterResetWithoutIntermediateUserTopic() throws Exception {
+        appID = testId + "-from-scratch";
+        STREAMS_CONFIG.put(StreamsConfig.APPLICATION_ID_CONFIG, appID);
 
         // RUN
-        KafkaStreams streams = new KafkaStreams(setupTopologyWithoutIntermediateUserTopic(), streamsConfiguration);
+        KafkaStreams streams = new KafkaStreams(setupTopologyWithoutIntermediateUserTopic(), STREAMS_CONFIG);
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java]: should we wrap the processing etc in `try{..}finally{..}` so we ensure that we do `streams.close()`? Also maybe close with a timeout?
Same elsewhere

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java [hunk 3]
```diff
@@ -42,31 +38,34 @@
...
-        CLUSTER = new EmbeddedKafkaCluster(NUM_BROKERS, props);
-        cluster = CLUSTER;
+        brokerProps.put(KafkaConfig$.MODULE$.ConnectionsMaxIdleMsProp(), -1L);
+        CLUSTER = new EmbeddedKafkaCluster(1, brokerProps);
     }
 
     @AfterClass
-    public static void globalCleanup() {
-        afterClassGlobalCleanup();
+    public static void afterClass() {
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java]: Can we move the `@AfterClass` annotation to the super class?

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationWithSslTest.java [hunk 4]
```diff
@@ -40,54 +37,55 @@
...
+            throw new RuntimeException(e);
+        }
+
+        CLUSTER = new EmbeddedKafkaCluster(1, brokerProps);
     }
 
     @AfterClass
-    public static void globalCleanup() {
-        afterClassGlobalCleanup();
+    public static void afterClass() {
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationWithSslTest.java]: as above

**Review by dguy — COMMENTED** [review]
Thanks @guozhangwang, left a few comments

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java (hunk 3 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java]: Ack.

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java]: Ack.

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java]: Ack.

**Review by dguy — APPROVED** [review]
Thanks @guozhangwang, LGTM

**Pull request merged**